### ISSUE TITLE
Using same bam file difference in bam track load time 

### ISSUE THREAD

**Issue by subji** [igvteam/igv.js#183]
hi,

I am currently using igv-0.9.3.js.

I use the same bam file, different for each gene.

There is a difference in function calls and function call time count.

In the latter case , call the function, tens of thousands of times .

![packalignmentrows_1](https://cloud.githubusercontent.com/assets/8937254/14664560/1a515882-0707-11e6-9bcd-54597c53f092.PNG)
![packalignmentrows_2](https://cloud.githubusercontent.com/assets/8937254/14664562/1a74416c-0707-11e6-9c71-3ca1adfd074c.PNG)

Why do I wonder if it makes a difference.

**Comment by jrobinso** [comment]
Please try against the latest check-in,  or the beta 
http://igv.org/web/beta/.   If the problem persists we will need access 
to a bam file and instructions (step by step) to reproduce the problem.

**Comment by subji** [comment]
when i used igv-beta.js occured error : 

> TypeError : Cannot read property 'chr' of undefined.


and my config of source code is 

> var config 	 = {
> 		      'showNavigation'		: true,
> 		      // 'genome'						: "hg19",
> 		     	'locus' 						: (chr + bp_str),
> 		     	'flanking'					: 5,
> 		      'tracks'						: [
> 		         {
> 		            'url' 		 		: bam_prefix + 'Sample3.bam',
> 		            'indexURL'		: bam_prefix + 'Sample3.bam.bai',
> 		            'label' 	 		: "Sample3",
> 		            'type' 		 		: 'bam',
> 		            'height' 			: 360,
> 		            'visibilityWindow' : '50000',
> 		         }
> 		      ],
> 		      'reference' 				: {
> 		      	'fastaURL' 				: seq_prefix + 'hg19.fasta',
> 		      	'cytobandURL'			: seq_prefix + 'cytoBand.txt'
> 		      }
> 			  };

**Comment by subji** [comment]
example.

i make simple jsfiddle code

http://jsfiddle.net/subji/ymxkz1kh/12/

**Comment by jrobinso** [comment]
I can't reproduce your error.  Could you try the html below?  It is based on your config using a public bam

`<!DOCTYPE html>

<html lang="en">
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">
    <link rel="shortcut icon" href="//igv.org/web/img/favicon.ico">
    <title>Integrative Genomics Viewer - BAM Example</title>

    
    <link rel="stylesheet" type="text/css"
          href="//ajax.googleapis.com/ajax/libs/jqueryui/1.11.2/themes/smoothness/jquery-ui.css"/>

    
    <link rel="stylesheet" type="text/css"
          href="//maxcdn.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css"/>

    
    <link rel="stylesheet" type="text/css" href="//igv.org/web/beta/igv-beta.css">

    
    <script type="text/javascript" src="//ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
    <script type="text/javascript" src="//ajax.googleapis.com/ajax/libs/jqueryui/1.11.2/jquery-ui.min.js"></script>

    
    <script type="text/javascript" src="//igv.org/web/beta/igv-beta.js"></script>


</head>

<body>

<div id="myDiv"></div>

<script type="text/javascript">

    $(document).ready(function () {
        var div,
                config,
                browser;

        div = $("#myDiv")[0];


         config = {
            'showNavigation' : true,
             'locus': '8:128,748,310-128,753,685',
            'flanking' : 5,
            'tracks': [
                {
                    'url' : 'https://data.broadinstitute.org/igvdata/BodyMap/hg19/IlluminaHiSeq2000_BodySites/brain_merged/accepted_hits.bam',
                    'indexURL' : 'https://data.broadinstitute.org/igvdata/BodyMap/hg19/IlluminaHiSeq2000_BodySites/brain_merged/accepted_hits.bam.bai',
                    'label' : "Sample3",
                    'type' : 'bam',
                    'height' : 360,
                    'visibilityWindow' : '50000'
                }

            ],
            'reference': {
                id: "hg19",
                fastaURL: "//igv.broadinstitute.org/genomes/seq/1kg_v37/human_g1k_v37_decoy.fasta",
                cytobandURL: "//igv.broadinstitute.org/genomes/seq/b37/b37_cytoband.txt"
            }
        };
        browser = igv.createBrowser(div, config);
    })


</script>

</body>

</html>

`

**Comment by jrobinso** [comment]
Sorry I missed your "fiddle" before.  I can reproduce the problem now, working on it.

**Comment by jrobinso** [comment]
Your bam file has some alignments that are marked  "paired",  but don't have a mate.   I think this is a mal-formed bam file,  but nevertheless added a check and these are considered un-paired.  Your fiddle example works now, you need to reload it.   This "beta" version will be released soon as 1.0.  Thanks for the report, and the fiddle, very helpful.

**Comment by subji** [comment]
Thank you.

that's good works now.
